pilot/pkg/serviceregistry/serviceentry/store_test.go

```diff
@@ -184,7 +199,42 @@ func TestServiceInstancesForDnsRoundRobinLB(t *testing.T) {
 		hostname:  "example.com",
 		namespace: "dns",
 	})
+	gotInstances = append(gotInstances, gotInstances2...)
 	if !reflect.DeepEqual(gotInstances, expected) {
 		t.Errorf("got unexpected instances : %v", gotInstances)
 	}
 }
+
+func TestUpdateInstances(t *testing.T) {
+	store := serviceInstancesStore{
+		ip2instance:            map[string][]*model.ServiceInstance{},
+		instances:              map[instancesKey]map[configKey][]*model.ServiceInstance{},
+		instancesBySE:          map[types.NamespacedName]map[configKey][]*model.ServiceInstance{},
+		instancesByHostAndPort: sets.Set[hostPort]{},
+	}
+	expectedInstances := []*model.ServiceInstance{
+		makeInstance(selector, []string{"1.1.1.1"}, 444, selector.Spec.(*networking.ServiceEntry).Ports[0], nil, PlainText),
+		makeInstance(selector, []string{"1.1.1.1", "2001:1::1"}, 445, selector.Spec.(*networking.ServiceEntry).Ports[1], nil, PlainText),
+	}
+	cKey := configKey{
+		namespace: "default",
+		name:      "test-wle",
+	}
+	store.addInstances(cKey, expectedInstances)
+	gotInstances := store.getAll()
+	if !reflect.DeepEqual(gotInstances, expectedInstances) {
+		t.Errorf("got unexpected instances : %v", gotInstances)
+	}
+
+	// 4. test update instances
+	expectedSeInstances := []*model.ServiceInstance{
+		makeInstance(selector, []string{"2.2.2.2", "2001:1::2"}, 444, selector.Spec.(*networking.ServiceEntry).Ports[0], nil, PlainText),
+		makeInstance(selector, []string{"2.2.2.2"}, 445, selector.Spec.(*networking.ServiceEntry).Ports[1], nil, PlainText),
+	}
+	store.updateInstances(cKey, expectedSeInstances)
```

Comment:
i think you should call update with just a single instance having single IP address.

Reply:
it must be something like:

```
	expectedInstances := []*model.ServiceInstance{
		makeInstance(selector, []string{"1.1.1.1"}, 444, selector.Spec.(*networking.ServiceEntry).Ports[0], nil, PlainText),
	}
	cKey := configKey{
		namespace: "default",
		name:      "test-wle",
	}
	store.addInstances(cKey, expectedInstances)
	// 4. test update instances
	expectedSeInstances := []*model.ServiceInstance{
		makeInstance(selector, []string{"1.1.1.1", "2001:1::2"}, 444, selector.Spec.(*networking.ServiceEntry).Ports[0], nil, PlainText),
	}
	store.updateInstances(cKey, expectedSeInstances)
```

Reply:
And you need to verify, the old stale entry(with single ip 1.1.1.1) does not remain after update.

Reply:
done, each check will get all instances by using `getAll()`, and there always is only one instance before and after the update.